File: common/src/main/java/generations/gg/generations/core/generationscore/world/item/GenericChestBlockItem.java
package generations.gg.generations.core.generationscore.world.item;

import generations.gg.generations.core.generationscore.world.level.block.generic.GenericChestBlock;
import net.minecraft.world.item.BlockItem;

public class GenericChestBlockItem extends BlockItem {
    public <T extends GenericChestBlock> GenericChestBlockItem(T arg, Properties arg2) {
        super(arg, arg2);
    }

//    @Override TODO: figure this out
//    public void initializeClient(Consumer<IClientItemExtensions> consumer) {
//        consumer.accept(new IClientItemExtensions() {
//            @Override
//            public BlockEntityWithoutLevelRenderer getCustomRenderer() {
//                return new GenericChestItemStackRenderer(Minecraft.getInstance().getBlockEntityRenderDispatcher(), Minecraft.getInstance().getEntityModels(), () -> new GenericChestBlockEntity(BlockPos.ZERO, getBlock().defaultBlockState()));
//            }
//        });
//    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/item/GenericChestBlockItem.kt
package generations.gg.generations.core.generationscore.common.world.item

import generations.gg.generations.core.generationscore.common.world.level.block.GenerationsBlockItem
import net.minecraft.core.Holder
import net.minecraft.world.level.block.Block
import net.minecraft.world.level.block.Blocks

class GenericChestBlockItem(arg: Holder<out Block>, arg2: Properties) : GenerationsBlockItem(arg, arg2) {
    //    @Override TODO: figure this out
//    public void initializeClient(Consumer<IClientItemExtensions> consumer) {
//        consumer.accept(new IClientItemExtensions() {
//            @Override
//            public BlockEntityWithoutLevelRenderer getCustomRenderer() {
//                return new GenericChestItemStackRenderer(Minecraft.getInstance().getBlockEntityRenderDispatcher(), Minecraft.getInstance().getEntityModels(), () -> new GenericChestBlockEntity(BlockPos.ZERO, getBlock().defaultBlockState()));
//            }
//        });
//    }
}
